File: 프로그래머스/2/42584. 주식가격/주식가격.py
from collections import deque 

def solution(prices):
    answer = []
    prices_que = deque(prices)
    
    while prices_que:
        price = prices_que.popleft()
        cnt = 0
        
        for p in prices_que:
            cnt += 1
            if price > p:
                break
        answer.append(cnt)
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        int cur;
        int cnt;
        int idx = 0;
        Deque<Integer> que = new ArrayDeque<>();
        for (int i : prices) que.add(i);
        while (!que.isEmpty()) {
            cur = que.poll();
            cnt = 0;
            for (int q : que) {
                cnt++;
                if (cur > q) break;
            }
            answer[idx++] = cnt;
        }
        return answer;
    }
}